docs/topics/gradle.md

```diff
@@ -641,3 +812,91 @@ When using [Gradle Kotlin DSL](https://github.com/gradle/kotlin-dsl), apply the
+
+## Kotlin daemon and its usage with Gradle
+
+The Kotlin daemon:
+* Runs along with Gradle daemon to compile the project.
+* Runs separately when you compile the project with IntelliJ IDEA capabilities.
+
+The Kotlin daemon starts at the Gradle [execution stage](https://docs.gradle.org/current/userguide/build_lifecycle.html#sec:build_phases)
+when some Kotlin compile task starts compiling the sources. The Kotlin daemon stops along with the Gradle daemon or after two idle hours with no Kotlin compilation.
+
+The Kotlin daemon uses the same JDK the Gradle daemon does.
+
+### Setting Kotlin daemon’s JVM arguments
+
+In the following list, the latter options override the previous ones:
+* If nothing is specified, the Kotlin daemon inherits arguments from the Gradle daemon. 
+  For example, in the `gradle.properties` file:
+
+    ```kotlin
+    org.gradle.jvmargs=-Xmx1500m,-Xms=500m
+    ```
+
+* If Gradle daemon has `kotlin.daemon.jvm.options` – use it in the `gradle.properties` file:
+
+    ```kotlin
+    org.gradle.jvmargs=-Dkotlin.daemon.jvm.options=-Xmx1500m,-Xms=500m
+    ```
+
+* You can add the`kotlin.daemon.jvmargs` property in the `gradle.properties` file:
+
+    ```kotlin
+    kotlin.daemon.jvmargs=-Xmx1500m,-Xms=500m
+    ```
+
+* You can specify arguments in `kotlin` extension:
+
+    <tabs>
+
+    ```groovy
+    kotlin {
+        kotlinDaemonJvmArgs = ["-Xmx486m", "-Xms256m", "-Duser.country=DE"]
+    }
+    ```
+
+    ```kotlin
+    kotlin {
+        kotlinDaemonJvmArgs = listOf("-Xmx486m", "-Xms256m", "-Duser.country=DE")
+    }
+    ```
+
+    </tabs>
+
+* You can specify arguments for a specific task:
+
+    <tabs>
+
+    ```groovy
+    tasks
+            .matching {
+                it.name == "compileKotlin" && it instanceof CompileUsingKotlinDaemon
+            }
+            .configureEach {
+                kotlinDaemonJvmArguments.set(["-Xmx1g", "-Xms512m"])
+            }
+    ```
+
+    ```kotlin
+    tasks
+        .matching { it.name == "compileKotlin" && it is CompileUsingKotlinDaemon }
+        .configureEach {
+            (this as CompileUsingKotlinDaemon).kotlinDaemonJvmArguments.set(listOf("-Xmx486m", "-Xms256m", "-Duser.country=DE"))
+        }
+    ```
+
+    </tabs>
+
+When configuring the Kotlin daemon's JVM arguments, please, note that:
+
+* It is expected to have multiple instances of the Kotlin daemon running at the same time when different subprojects or tasks have different sets of JVM arguments.
+* New Kotlin daemon instance starts only when Gradle runs a related compilation task and existing Kotlin daemons do not have the same set of JVM arguments.
+  Imagine, your project has a lot of subprojects: most of them require some heap memory for Kotlin daemon but one module requires a lot (though it is compiled rarely).
+  In such situation, it makes sense to provide a different set of JVM arguments for such a module, so a new Kotlin daemon could start.
```

Comment:
Maybe slightly rephrase `so a new Kotlin daemon could start.` into `so Kotlin daemon with more heap size will only be started for developers touching this specific module`.